storagemarket/provider.go

```diff
@@ -271,65 +284,80 @@ func (p *Provider) ExecuteDeal(dp *types.DealParams, clientPeer peer.ID) (pi *ap
 		}
 	}
 
-	// if it is an offline deal  just persist it in the DB here and return
+	// if it is an offline deal just persist it in the DB here and return
 	if dp.IsOffline {
+		p.dealLogger.Infow(dp.DealUUID, "offline deal accepted, waiting for data import")
+
 		ds.CreatedAt = time.Now()
 		ds.Checkpoint = dealcheckpoints.Accepted
 		if err := p.dealsDB.Insert(p.ctx, &ds); err != nil {
 			return nil, nil, fmt.Errorf("failed to insert deal in db: %w", err)
 		}
+
+		// Set up pubsub for deal updates
+		dh, _ := p.mkAndInsertDealHandler(dp.DealUUID)
+		pub, err := dh.bus.Emitter(&types.ProviderDealState{}, eventbus.Stateful)
+		if err != nil {
+			err = fmt.Errorf("failed to create event emitter: %w", err)
+			p.failDeal(pub, &ds, err)
+			p.cleanupDealLogged(&ds)
+			return nil, nil, err
+		}
+
+		// publish "new deal" event
+		p.fireEventDealNew(&ds)
+		// publish an event with the current state of the deal
+		p.fireEventDealUpdate(pub, &ds)
+
 		return &api.ProviderDealRejectionInfo{Accepted: true}, nil, nil
 	}
 
-	// setup clean-up code
-	var tmpFile filestore.File
-	var dh *dealHandler
-	cleanup := func() {
-		if tmpFile != nil {
-			_ = os.Remove(string(tmpFile.OsPath()))
+	dh, _ := p.mkAndInsertDealHandler(dp.DealUUID)
```

Comment:
Why do we need the refactoring from here to the end of the function ? What problem does it solve ? Dosen't have anything to do with an offline deal.

Reply:
Please can we get rid of the refactoring below. If it adds some benefit, we can add it in a separate PR for refactoring code. But, let's not increase the scope of this offline deal PR if it's okay with you.